CRM/Contact/Form/Search/Criteria.php

```diff
@@ -35,6 +35,32 @@ class CRM_Contact_Form_Search_Criteria {
    * @param CRM_Core_Form $form
    */
   public static function basic(&$form) {
+    // Allows the template to determine if another template exists:
+    $smarty = CRM_Core_Smarty::singleton();
```

Comment:
This didn't work for me. What did work was

a) adding a new file called modifier.template_exists.php in packages/Smarty/plugins
b) then clearing smarty caches

I tried using 
https://gitlab.athlonsofia.com/athlon/silla.io/blob/3b669c42e34d72d62f4ac624244d979402c7a98e/resources/smarty_plugins/modifier.template_exists.php

but it didn't work for me so am using 

```
function smarty_modifier_template_exists($template)
{

  $templatePath = __DIR__ . '/../../../templates/' . $template;

  return is_file($templatePath);
}

```

If you want to put up a PR for a new modifer against the packages repo I will merge it